### DISCUSSION TITLE
reload_extension seems not to reload the imported files

### DISCUSSION THREAD

**Discussion by xavyc** [Rapptz/discord.py#9051] [Q&A]
Hello, 

I am currently working on rewriting my bot, while trying to make the extensions "cleaner" in my opinion I came across a behavior that I do not really understand. 

With the view and modal, I wanted to be able to separate them in separate files and so I imported them into my Cog.

But the problem is that when I try to reload one of these extensions it seems that the imported files are not reloaded.

How can I do to force the re-import of these files? 

my cogs folder looks like this :

├── backend
├── cogs
│   ├── plugin_name
│   │   ├── plugin.py
│   │   ├── view.py
│   │   ├── other_view.py

here the changes of plugin.py are loaded but not those of view.py and other_view.py

Thanks in advance

**Comment by Rapptz** [comment]
The way Python imports work is that they're executed once and then cached so subsequent imports are cheap. The library maintains ownership of the imports for the extensions (in this case, `plugin.py`) but if `plugin.py` imports `view.py` and `other_view.py` then those aren't going to be reloaded by the library and it'll be the cached ones instead.

Working with these types of dependencies tends to be complicated in the general case. In Python you can use [`importlib.reload`](https://docs.python.org/3/library/importlib.html#importlib.reload) to reload a pre-existing module. You can find modules inside [`sys.modules`](https://docs.python.org/3/library/sys.html#sys.modules). So, if you wanted to reload the `foo.bar` module you'd do:

```py
import sys, importlib
sys.modules['foo.bar'] = importlib.reload(sys.modules['foo.bar'])
```

Note that you need to do this *before* reloading the extension for your changes to take effect.

Getting the modules that you need to reload is a bit complicated, and that depends entirely on how your program is structured. I'll leave that bit up to you. Hopefully these crumbs are enough to let you put together the entire piece.

**Reply by xavyc** [reply]
I see that make more sense now

I managed to implement it in my own reload function.

Thanks!
